### PR TITLE
fix: high-contrast style in Button, AccentButton, ActionToggle, ActionTrigger, Call to Action, NeutralButton, LightweightButton, OutlineButton, and StealthButton to support Edge icon components and add more common constants to high-contrast utility file.

### PR DESCRIPTION
**Pull request by khamudom**
# Description
While working on high contrast in Edge Web UI, I was noticing the icon components, from our icon packages, were not changing colors on hover. The fix was to set its fill property to the correct color on hover. We should not have to do this. The components should take care of setting the correct colors already.
After investigating, the `fill` property, in our button styles, was not set on hover. During the initial high contrast pass, we were testing to make sure text and icons were working properly, which they did, but did not test to see if icons from edge-icons or news-icons packages worked. If I had imported an edge-icon into a storybook example, I would have noticed that the fill property should have also been set on hover.
The fix is to set, as an example, `fill: "HighlightText"` (or whatever color it should be) to be included in the hover state. Now when any icon is being used in our Buttons, whether it is a beforeContent, afterContent, or an imported icon from one of our icon packages, the high contrast colors will be correct in all states.

There were also more common patterns that were be added to the `high-contrast` utility file.
Added:
- highContrastAccent
- highContrastSelectionOutline
- highContrastSelectionForeground
- highContrastSelectionBackground

The Button storybook example was also updated. The Stealth example was incorrect.

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [x] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-components-styles-msft/src/utilities/high-contrast.ts [hunk 1]
```diff
@@ -94,6 +105,15 @@ export const highContrastDoubleFocus: CSSRules<DesignSystem> = {
...
 export const highContrastSelection: CSSRules<DesignSystem> = {
+    [highContrastSelector]: {
+        background: HighContrastColor.selectedBackground,
+        color: HighContrastColor.selectedText,
+        fill: HighContrastColor.selectedText,
+    },
+};
+
+// Used to set 'selected-text' color with an outline
+export const highContrastSelectionOutline: CSSRules<DesignSystem> = {
```
Comment (bheston) [inline packages/fast-components-styles-msft/src/utilities/high-contrast.ts]: Looks like I missed one last time, but "selection" should be "selected". Can we rename the new ones and deprecate/alias the other since it's already been published. "Selection" sounds more like highlighted text or a single item in a list. "Selected" is a state like we're using it here. That will coordinate better with a new feature I'm about to do for non-HC selected colors.

File: packages/fast-components-styles-msft/src/button/index.ts [hunk 2]
```diff
@@ -89,6 +94,7 @@ const applyTransparentBackplateStyles: CSSRules<DesignSystem> = {
         [highContrastSelector]: {
             background: "ButtonFace !important",
             color: "ButtonText !important",
+            fill: "Highlight",
```
Comment (bheston) [inline packages/fast-components-styles-msft/src/button/index.ts]: Shouldn't the icon fill be the same color as the foreground text?

File: packages/fast-components-styles-msft/src/button/index.ts [hunk 3]
```diff
@@ -179,6 +169,7 @@ const styles: ComponentStyles<ButtonClassNameContract, DesignSystem> = {
                 background: "HighlightText !important",
```
Comment (bheston) [inline packages/fast-components-styles-msft/src/button/index.ts]: Should this whole thing be `highContrastSelection` maybe with `border-color` added on? Or should the border always be a part of this combo?

File: packages/fast-components-styles-msft/src/utilities/high-contrast.ts [hunk 4]
```diff
@@ -110,6 +130,14 @@ export const highContrastForeground: CSSRules<DesignSystem> = {
     },
 };
 
+// Used to set foreground and glyph to be 'select-text' color
+export const highContrastSelectionForeground: CSSRules<DesignSystem> = {
+    [highContrastSelector]: {
+        color: "HighlightText !important",
```
Comment (bheston) [inline packages/fast-components-styles-msft/src/utilities/high-contrast.ts]: Can all values in this file use the constants? Maybe add `important` to each color anyway.

File: packages/fast-components-styles-msft/src/button/index.ts (hunk 2 above)
Comment (khamudom) [reply packages/fast-components-styles-msft/src/button/index.ts]: This is a one off situation. in our Lightweight and Justified buttons, the beforeContent and afterContent do not get an underline indicator when a user hovers over it. It does slightly change color. I had to separate the color from the icon and the text, because on hover the text does not change, we just add an underline. If both the icon and text are the same color, and, they are both together, its fine when the user hovers, because you at least get an underline on the text. But if it is only the icon, and you hover over it, there is indication of hover., because the icon is is the same color in rest and hover state.
As I type this, I am thinking I can just change `color` to "Highlight", that should work.

Comment (bheston) [reply packages/fast-components-styles-msft/src/button/index.ts]: Sounds good. Maybe add a comment. That file is a little hard to parse because of those types of situations.

File: packages/fast-components-styles-msft/src/utilities/high-contrast.ts (hunk 4 above)
Comment (khamudom) [reply packages/fast-components-styles-msft/src/utilities/high-contrast.ts]: create a function called `ImportantColor` to use here

File: packages/fast-components-styles-msft/src/button/index.ts (hunk 3 above)
Comment (khamudom) [reply packages/fast-components-styles-msft/src/button/index.ts]: put them in a const called `highContrastSelectedOutline`

File: packages/fast-components-styles-msft/src/button/index.ts (hunk 2 above)
Comment (khamudom) [reply packages/fast-components-styles-msft/src/button/index.ts]: updated

File: packages/fast-components-styles-msft/src/utilities/high-contrast.ts (hunk 1 above)
Comment (khamudom) [reply packages/fast-components-styles-msft/src/utilities/high-contrast.ts]: updated to be `selected`

**Comment by janechu** [comment]
Can you change the title to "fix: add missing style properties to Button, AccentButton (etc.)" and fill in the etc. with all relevant components?

**Comment by janechu** [comment]
The language is still off, we want the language to read more like what this is accomplishing, so "adds missing style properties to [list of affected components]".

When this is added to the changelog it will go under fixes with this line so it should be as descriptive as possible in what it is doing. I would also leave out "our", this doesn't make sense given the context (as a means of communication to developers taking a dependency on our package).

File: packages/fast-components-styles-msft/src/utilities/high-contrast.ts [hunk 5]
```diff
@@ -16,18 +16,22 @@ export enum HighContrastColor {
     background = "Background",
 }
 
+function ImportantColor(color: HighContrastColor): string {
+    return color + "!important";
```
Comment (janechu) [inline packages/fast-components-styles-msft/src/utilities/high-contrast.ts]: no space between the color and !important, also prefer using back ticks here.

File: packages/fast-components-styles-msft/src/utilities/high-contrast.ts [hunk 6]
```diff
@@ -134,3 +174,17 @@ export const highContrastBackground: CSSRules<DesignSystem> = {
         background: HighContrastColor.buttonText,
     },
 };
+
+// Used to set background to be 'selected-text' color
+export const highContrastSelectionBackground: CSSRules<DesignSystem> = {
```
Comment (janechu) [inline packages/fast-components-styles-msft/src/utilities/high-contrast.ts]: where is this getting used?

File: packages/fast-components-styles-msft/src/utilities/high-contrast.ts [hunk 7]
```diff
@@ -93,20 +108,45 @@ export const highContrastDoubleFocus: CSSRules<DesignSystem> = {
 };
 
 // Used to set 'selected-text' color
-export const highContrastSelection: CSSRules<DesignSystem> = {
+export const highContrastSelected: CSSRules<DesignSystem> = {
```
Comment (janechu) [inline packages/fast-components-styles-msft/src/utilities/high-contrast.ts]: this looks like a breaking change :bangbang:

File: packages/fast-components-styles-msft/src/utilities/high-contrast.ts (hunk 5 above)
Comment (khamudom) [reply packages/fast-components-styles-msft/src/utilities/high-contrast.ts]: updated with back ticks

File: packages/fast-components-styles-msft/src/utilities/high-contrast.ts (hunk 6 above)
Comment (khamudom) [reply packages/fast-components-styles-msft/src/utilities/high-contrast.ts]: this is now being used

File: packages/fast-components-styles-msft/src/utilities/high-contrast.ts (hunk 7 above)
Comment (khamudom) [reply packages/fast-components-styles-msft/src/utilities/high-contrast.ts]: brought back the const and add deprecated comment

Comment (janechu) [reply packages/fast-components-styles-msft/src/utilities/high-contrast.ts]: if this is just a name change you can make the deprecated version simply point to the new version, like so:
`export const highContrastSelection: CSSRules<DesignSystem> = highContrastSelected;`

File: packages/fast-components-styles-msft/src/utilities/high-contrast.ts [hunk 8]
```diff
@@ -16,18 +16,22 @@ export enum HighContrastColor {
     background = "Background",
 }
```
Comment (janechu) [inline packages/fast-components-styles-msft/src/utilities/high-contrast.ts]: I can't comment on it (because github UI doesn't allow me) but the disabledText is marked with an !important, given the new Important utility, should you remove it here and add it in like you are doing with the rest of these?

File: packages/fast-components-styles-msft/src/utilities/high-contrast.ts [hunk 9]
```diff
@@ -16,18 +16,22 @@ export enum HighContrastColor {
...
 
+function ImportantColor(color: HighContrastColor): string {
```
Comment (nicholasrice) [inline packages/fast-components-styles-msft/src/utilities/high-contrast.ts]: typically capitol letters are reserved for class constructors, enums, interfaces, and type aliases - can you rename to `importantColor`?

You should also consider abstracting - the only assumption this code really makes is that the argument type is a string. You could move this to a generic utility file and change the type signature to simply expect a string value. Then this could be used to add important to any css value

Comment (khamudom) [reply packages/fast-components-styles-msft/src/utilities/high-contrast.ts]: abstract the function to its own utility

File: packages/fast-components-styles-msft/src/utilities/high-contrast.ts (hunk 8 above)
Comment (khamudom) [reply packages/fast-components-styles-msft/src/utilities/high-contrast.ts]: updated

File: packages/fast-components-styles-msft/src/utilities/important.ts [hunk 10]
```diff
@@ -0,0 +1,6 @@
+/**
+ * Sets a css value to be '!important'.
+ */
+export function importantValue(value: string): string {
```
Comment (bheston) [inline packages/fast-components-styles-msft/src/utilities/important.ts]: Since this is intended for style sheets, and most style sheet functions are DesignSystemResolvers, can you change this to `value: string | DesignSystemResolver<string>` and then call `checkDesignSystemResolver(value)`?

Comment (khamudom) [reply packages/fast-components-styles-msft/src/utilities/important.ts]: updated. thanks Brian

File: packages/fast-components-styles-msft/src/utilities/high-contrast.ts (hunk 7 above)
Comment (khamudom) [reply packages/fast-components-styles-msft/src/utilities/high-contrast.ts]: updated
